### DISCUSSION TITLE
VST3 Support Only?

### DISCUSSION THREAD

**Discussion by Xpktro** [monocasual/giada#496] [Q&A] [answered]
I've been using Giada for a while now and wanted to use it for giving an entry-level music workshop so I was looking for free multi-platform VST to use.  

Given that most VST alternatives are still in v2 (.dll/vst), I found that only VST3 (.vst3) plugins are detected/supported (I'm using the latest Windows build). Is this intentional? Are there any flag/patch/setting that can be used to support them? I wouldn't want my choices to be that restricted :(

**Comment by Xpktro** [comment]
P.S. I've been testing Giada v0.16.4 @ macOS and for some reason I'm only getting VST plugins detected and not a single VST3, is this normal? 😓

**Comment by gvnnz** [comment]
Hi @Xpktro , yes unfortunately we ship VST3-only binaries, since VST2.x has been deprecated by Steinberg.  You can try build Giada yourself from source, it's not hard! https://www.giadamusic.com/documentation-compiling-from-source The tricky part is to find the old VST2.x SDK. If my memory serves me right it should be still included in the official VST3 SDK.

**Reply by Xpktro** [reply]
Got it, I also found that deprecation notice while researching about the issue. However macOS Giada is not detecting my VST3 plugins (only vsts) so that got me puzzled thinking it had something to do with enabling any flag at compile time or something.

**Reply by CTBlaze** [reply]
Ciao Giovanni, please keep observing the market - I think it's too early to omit on VST2, too many great mainstream plugins out there which are still not available in VST3.  
Could you PLEASE consider providing a precompiled binary with BOTH VST3 and VST2 support, I would also pay for your time/effort, if neccessary!
BTW, do you see issues with performance, or violation with any Steinberg licensing?  why your recommendation against VST2 compilation?
 (grr, why did the audio industry fall into this proprietary hellhole decades ago? because WIndows did not provide anything professional audio, I know. By now it does, but NOBODY is using it!)
MIlle Grazie

**Reply by CTBlaze** [reply]
OMG, I found this thread, I did not even know about the details of VST2 debacel... Steinberg sucks.
[https://forum.juce.com/t/hosting-vst2-plugins-under-vst3-sdk-license/37079](https://forum.juce.com/t/hosting-vst2-plugins-under-vst3-sdk-license/37079) 
also read the newer replies...
Giovanni, has monocausal signed a VST2 license agreement in the past?  Then it's no problem to provide binaries with VST2 hosting support, it seems.

**Reply by CTBlaze** [reply]
And finally, has either of you tried compiling Giada with the FST headers, for VST2 support? see here:
[https://git.iem.at/zmoelnig/FST/](https://git.iem.at/zmoelnig/FST/)

**Comment by Xpktro** [comment, marked as answer]
After some days researching about the project and its capabilities I got my answers and for every curious individual thay may have the same questions as me:

* Giada supports VST3 and * can * support VST2
* VST2 support depends on you building giada manually from source while adding a copy of the VST 2.x SDK (which is no longer supported by Steinberg/officially distributed)
* Current pre-built releases have various degree of OS/VST compatibility. 

So although it's possible to have a *current* Giada version with VST2 compatibility, you'll have to make it yourself.

**Reply by CTBlaze** [reply]
thank you for the summary. It's a bit like the situation with Adobe FLash - which had been declared deprecated for 10+ years; 
now Steinberg (with a quasi monoply on plugin technology in WIndows world, which is DANGEROUS) tries to enforce / speed up the death of VST2 plugins, yet SO many really good products are still being provided as VST2 ONLY, even by ginats like Native Instruments. Apparently, thos have become too lame in their product lifecycle processes, not evben their killer product Reactor or Guitar Rig are availabe in VST3 ;(
Do you have a hint on where to DL the VST2 SDK, please? Or, would you share the binary you compiled?

**Reply by Xpktro** [reply]
Hello! the Internet Archive has a functional VST2 SDK copy. A simple web search should take you to the right place :)

**Reply by CTBlaze** [reply]
Gracias - can you compile it to support both, VST2 & VST3?

**Reply by CTBlaze** [reply]
Ola again, say, have you ever tried compiling it with the GPL based FST headers? see  [https://git.iem.at/zmoelnig/FST/ ](https://git.iem.at/zmoelnig/FST/ )

**Comment by gvnnz** [comment]
@CTBlaze et al.: unfortunately we have no plans to going back to support VST2.x here. Leaving aside the legal issues you guys mentioned on this thread, the maintenance burden would be too high on our side. 

The only way is to find the old VST2.x SDK somewhere on the Internet and compile Giada from source. The annoying part is to find the SDK (no idea where to look honestly), compiling Giada is kind of straightforward ([instructions here](https://www.giadamusic.com/documentation-compiling-from-source)). 

FST: I'm not familiar with those headers, it would be great if you could try and report back!

**Reply by CTBlaze** [reply]
hmm, I am reluctant to install VIsual Studio or similar on my new laptop, which is mainly for other things, like audio ;)
Also, I have not compiled things in WIndoze for decades, last time it was PHP CLI , I think. 
WHich command line compiler packages for WIndoews can you recommend?  MinGW 64? TMD?  DO I still need cmake on top?
Your instructions for compilation miss out on which compilers to use, unfortunately. 
PS: Microsoft also offers downlaod of command lien build tools, but they say you need a VS license (which may be the COmmunity one also, but you need to install it first, and I am not doing that). Maybe I do it on a Virtual Machine, if I find the time. 
BTW, the reverse engineering document by the author of FST seems VERY intereresting, esp. for audio techies like you, I recommend reading:    https://github.com/pierreguillot/FTS/blob/master/docs/REVERSE_ENGINEERING.md

**Reply by gvnnz** [reply]
> WHich command line compiler packages for WIndoews can you recommend? MinGW 64? TMD? DO I still need cmake on top?

Since you are on Windows, Visual Studio is the way to go. You can grab one of the virtual machines Microsoft provides for testing purposes [here](https://developer.microsoft.com/en-us/microsoft-edge/tools/vms/), Visual Studio should be already installed (if memory serves). CMake will take care of everything else.

Thanks for the link, pretty cool stuff in there!
